### PR TITLE
Create view for mentor application

### LINKED ISSUE #16: Create Apply as a mentor view
**Is your feature request related to a problem? Please describe.**
- Need to develop the "apply as a mentor view" on the admin dashboard.

**Describe the solution you'd like**
- Create a dummy UI adhering to the given view on the admin wireframes.

**Additional context**
- This is the view that needs to be developed, a dummy form would suffice.
- For additional details, please follow this link to the [wireframes thread](https://sef.discourse.group/t/designing-wireframes-for-scholarx-platform/148).

image

### PR DESCRIPTION
**Pull request by Gimhan-minion**
## Purpose
The purpose of this PR is to fix #16

## Goals
- To create a view for mentor Application

## Approach
- Used antd Steps component to create separate sections

## Screenshots
image

##  Checklist
- [x] This PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.
- [x] I have read and understood the development best practices guidelines ( http://bit.ly/sef-best-practices )
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/scenes/Home/components/MentorApplication/index.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+  };
+  const selectBack = () => {
+    setCurrent(current - 1);
+    console.log(current);
+  };
+  const selectDone = () => {
+    setCurrent(current + 1);
+    message.success('Application successfully submitted');
+    console.log(current);
+  };
```
Comment (Gravewalker666) [inline src/scenes/Home/components/MentorApplication/index.tsx]: Let's lose all the `console.logs`

**Review by Gravewalker666 — DISMISSED** [review]
Great work dude, requested some minor changes. Please look into it.

File: src/scenes/Home/components/MentorApplication/index.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+import logo from '../../scholarx.png';
+import styles from './styles.css';
+
+const { Step } = Steps;
+const { TextArea } = Input;
+
+function MentorApplication() {
+  const [current, setCurrent] = useState<number>(0);
+  const selectGo = () => {
+    setCurrent(current + 1);
```
Comment (jayasanka-sack) [inline src/scenes/Home/components/MentorApplication/index.tsx]:
```suggestion
    setCurrent(2);
```
We know that the next step's number is `2`. Therefore no need to do calculations.

File: src/scenes/Home/components/MentorApplication/index.tsx [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+import { Steps, Button, Row, Col, Input, message } from 'antd';
+import logo from '../../scholarx.png';
+import styles from './styles.css';
+
+const { Step } = Steps;
+const { TextArea } = Input;
+
+function MentorApplication() {
+  const [current, setCurrent] = useState<number>(0);
+  const selectGo = () => {
```
Comment (jayasanka-sack) [inline src/scenes/Home/components/MentorApplication/index.tsx]: The function name should reflect what the function does. 

Here's a great inspiration.
https://dmitripavlutin.com/coding-like-shakespeare-practical-function-naming-conventions/

File: src/scenes/Home/components/MentorApplication/index.tsx [hunk 4]
```diff
@@ -0,0 +1,121 @@
+import React, { useState } from 'react';
+import { Steps, Button, Row, Col, Input, message } from 'antd';
+import logo from '../../scholarx.png';
+import styles from './styles.css';
+
+const { Step } = Steps;
+const { TextArea } = Input;
+
+function MentorApplication() {
+  const [current, setCurrent] = useState<number>(0);
```
Comment (jayasanka-sack) [inline src/scenes/Home/components/MentorApplication/index.tsx]: change the variable name `current ` to something descriptive. ex: `currentStep`

File: src/scenes/Home/components/MentorApplication/index.tsx [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+        )}
+        {current == 3 && (
+          <Row>
+            <Col md={2} />
+            <Col md={12}>
+              <div className={styles.form}>
+                <Button href="/home">Go Back</Button>
+              </div>
+            </Col>
+          </Row>
```
Comment (jayasanka-sack) [inline src/scenes/Home/components/MentorApplication/index.tsx]: No need for this step at the moment.

File: src/scenes/Home/components/MentorApplication/styles.css [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    margin-top: 2rem;
+}
+
+.form{
+    margin-top: 3rem;
+    padding-top: 10px;
+    padding-left: 10px;
+}
+
+.customButtonNext{
```
Comment (jayasanka-sack) [inline src/scenes/Home/components/MentorApplication/styles.css]:
```suggestion
.nextButton{
```

File: src/scenes/Home/components/MentorApplication/index.tsx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+        )}
+        {current == 1 && (
+          <Row>
+            <Col md={2} />
+            <Col md={12}>
+              <div className={styles.form}>
+                <h3 className={styles.question}>
+                  <b>Q2.</b>Why did you apply for ScholarX?
```
Comment (jayasanka-sack) [inline src/scenes/Home/components/MentorApplication/index.tsx]: The second step is to add the instructions for the mentees.

**Review by jayasanka-sack — CHANGES_REQUESTED** [review]
Thanks @Gimhan-minion !
I've added some comments

File: src/scenes/Home/components/MentorApplication/index.tsx [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+                  <b>Q1.</b>Why did you apply as a Mentor?
+                </h3>
+                <TextArea rows={8} />
+                <Button onClick={selectGo} className={styles.customButtonNext}>
+                  Next
+                </Button>
+              </div>
+            </Col>
+          </Row>
+        )}
```
Comment (jayasanka-sack) [inline src/scenes/Home/components/MentorApplication/index.tsx]: There are no 3 questions. 
Let's discuss this at today's meeting.

File: src/scenes/Home/components/MentorApplication/styles.css [hunk 9]
```diff
@@ -0,0 +1,17 @@
+.container{
+    margin-top:5rem;
+}
+
+.logo {
+    width: 250px;
+    margin-bottom: 2rem;
+}
+
+.form{
```
Comment (jayasanka-sack) [inline src/scenes/Home/components/MentorApplication/styles.css]:
```suggestion
.form {
```

File: src/scenes/Home/components/MentorApplication/styles.css [hunk 10]
```diff
@@ -0,0 +1,17 @@
+.container{
```
Comment (jayasanka-sack) [inline src/scenes/Home/components/MentorApplication/styles.css]:
```suggestion
.container {
```

**Comment by Gimhan-minion** [comment]
Changes made @jayasanka-sack

**Review by jayasanka-sack — APPROVED** [review]
Thanks @Gimhan-minion ! 😍

**Pull request merged**